lib/transport/transport-udp-socket.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2002-2019 Balabit
+ * Copyright (c) 1998-2019 Balázs Scheidler
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+#include "transport-udp-socket.h"
+#include "transport/transport-socket.h"
+#include "scratch-buffers.h"
+#include "str-format.h"
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <errno.h>
+
+typedef struct _LogTransportUDP LogTransportUDP;
+struct _LogTransportUDP
+{
+  LogTransportSocket super;
+  GSockAddr *local_addr;
+};
+
+#if defined(SYSLOG_NG_HAVE_CTRLBUF_IN_MSGHDR)
+static void
+_feed_aux_from_cmsg(LogTransportUDP *self, LogTransportAuxData *aux, struct msghdr *msg)
+{
+  struct cmsghdr *cmsg;
+
+  for (cmsg = CMSG_FIRSTHDR(msg); cmsg != NULL; cmsg = CMSG_NXTHDR(msg, cmsg))
+    {
+
+#ifdef __FreeBSD__
+      if (cmsg->cmsg_level == IPPROTO_IP && cmsg->cmsg_type == IP_RECVDSTADDR)
+        {
+          gint cmsg_header_len = (gchar *) CMSG_DATA(cmsg) - (gchar *) cmsg;
+          struct sockaddr *sa = (struct sockaddr *) CMSG_DATA(cmsg);
```

Comment:
Could you extract the blocks inside the compiler switches to a separate functions? It would improve readibility.

Reply:
Sure, will do so.

Reply:
I've just pushed b22887c6f4b5efed365f192bc875e8d0d4e652b0 to address this issue.